test/interval_tests/construction.jl

```diff
@@ -196,3 +196,12 @@ end
     @test 3 ± 0.1 == Interval(2.9, 3.1)
     @test 0.5 ± 1 == Interval(-0.5, 1.5)
 end
+
+# issue 192:
+@testset "Disallow a single NaN in an interval" begin
+    a = Interval(NaN, 2)
+    @test isnai(a)
```

Comment:
Since `isnai(a)` verifies that either `a.lo` or `a.hi` is `NaN`, this tests does not quite guarantee that *both* are `NaN`s. I suggest adding another test of the form `@test isnan(a.lo) && isnan(a.hi)`, which precisely addresses this. Once said this, I don't think we need to change `isnai()`...